sui/wormhole/sources/resources/id_registry.move

```diff
@@ -0,0 +1,78 @@
+module wormhole::id_registry {
+    use wormhole::external_address::{Self, ExternalAddress};
+
+    /// Resource to keep track of an increasing `index` value, which is used to
+    /// generate a new `ExternalAddress`.
+    struct IdRegistry has store {
+        index: u64
+    }
+
+    public fun new(): IdRegistry {
+        IdRegistry { index: 0 }
```

Comment:
we should start at 1. thanks to solidity, 0 is no longer a usable number (and starting the id at 0 will make it impossible to do a registration on EVM chains as they perform a non-zero check)

Reply:
Ah, there shouldn't be anything issued by `IdRegistry` with the value `0` (which maybe I should remove the `index` getter to make this less confusing) because `next_address` upticks the `index` before generating an `ExternalAddress`.

Should I remove the public `index` getter? I've only used it in testing, so I can make it `#[test_only]`.

Reply:
Added [ebad8c9](https://github.com/wormhole-foundation/wormhole/pull/2494/commits/ebad8c92c79c9b0fdd88eb043f58d488d500cafa) so now only `new` and `next_address` are the only public methods.

Reply:
We can do it that way. I think the only reason was to refactor the registry to save a line of code when generating a new `ExternalAddress`. But we can keep the stored index as the next index that will be used.